transfocate/transfocator.py

```diff
@@ -180,3 +185,37 @@ def focus_at(self, value=None, wait=False, timeout=None, **kwargs):
         if wait:
             status_wait(status, timeout=timeout)
         return status
+
+
+class TransfocatorEnergyInterrupt(Exception):
+    """
+    Custom exception returned when input beam energy (user defined
+    or current measured value) changes significantly during
+    calculation
+    """
+    pass
+
+
+def constant_energy(func):
+    """
+    Ensures that requested energy does not change during calculation
+
+    Parameters:
+    transfocator_obj: transfocate.transfocator.Transfocator object
+    energy_type: string
+        input string specifying 'req_energy' or 'beam_energy'
+        to be monitored during calculation
+    """
+    @wraps(func)
+    def with_constant_energy(transfocator_obj, energy_type, *args, **kwargs):
+        try:
+            energy_signal = getattr(transfocator_obj, energy_type)
+        except:
```

Comment:
Let's be more specific here:
```suggestion
        except AttributeError:
```

Reply:
So when written this way I shouldn't raise any error in the following line, just print a message?